mlflow/utils/databricks_utils.py

```diff
@@ -798,18 +798,25 @@ def get_databricks_env_vars(tracking_uri):
     return env_vars
 
 
+class DatabricksRuntimeVersion(NamedTuple):
+    is_client_image: bool
+    major: int
+    minor: int
+
+
 def get_databricks_runtime_major_minor_version():
     dbr_version = os.environ["DATABRICKS_RUNTIME_VERSION"]
     try:
         dbr_version_splits = dbr_version.split(".", maxsplit=2)
-        # don't int-ify the major version, as "client" is a valid major key
-        major = (
-            int(dbr_version_splits[0])
-            if dbr_version_splits[0].isnumeric()
-            else dbr_version_splits[0]
-        )
-        minor = int(dbr_version_splits[1])
-        return major, minor
+        if dbr_version_splits[0] == "client":
+            is_client_image = True
+            major = int(dbr_version_splits[1])
+            minor = int(dbr_version_splits[2]) if len(dbr_version_splits) > 2 else 0
+        else:
+            is_client_image = False
+            major = int(dbr_version_splits[0])
+            minor = int(dbr_version_splits[1])
+        return DatabricksRuntimeVersion(is_client_image, major, minor)
```

Comment:
/ai Add a class method `parse` to `DatabricksRuntimeVersion` and move the version parsing logic there.